### ISSUE TITLE
generator-star ignores object literal shorthand methods

### ISSUE THREAD

**Issue by btmills** [eslint/eslint#1680]
This should fail due to the space between the asterisk and the method name:

```js
// "generator-star": [2, "end"]
({
    * method() {}
})
```

#1678 was merged because #1677 was more important than this case. That PR prevented generator-star from checking star position for shorthand methods because a parser fix or workaround will be required. This issue is blocked until a decision is made in eslint/espree#36 (and [Esprima issue 625](https://code.google.com/p/esprima/issues/detail?id=625)) regarding node ranges for shorthand methods.

**Comment by btmills** [comment]
The Esprima issue has been moved to jquery/esprima#1029.

**Comment by btmills** [comment]
The parser issue has been fixed with jquery/esprima#1036 and eslint/espree#61. The next release of Espree will unblock this issue.

**Comment by nzakas** [comment]
I'll plan on doing that this weekend.

**Comment by nzakas** [comment]
Espree has been upgraded, so this is unblocked.

**Comment by btmills** [comment]
The messages in this rule are confusing. For example:

```js
function *
